repos/spack_repo/builtin/packages/uwtools/package.py

```diff
@@ -0,0 +1,46 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.python import PythonPackage
+
+from spack.package import *
+
+
+class Uwtools(PythonPackage):
+    """UW Tools is a modern, open-source Python package that helps
+    automate common tasks needed for many standard numerical weather
+    prediction (NWP) workflows. It also provides drivers to automate
+    the configuration and execution of Unified Forecast System (UFS)
+    components, providing flexibility, interoperability, and
+    usability to various UFS Applications."""
+
+    homepage = "https://uwtools.readthedocs.io/en/stable/"
+    url = "https://github.com/ufs-community/uwtools/archive/refs/tags/v2.7.1.tar.gz"
+    git = "https://github.com/ufs-community/uwtools.git"
+
+    maintainers("NaureenBharwaniNOAA", "christinaholtNOAA", "elcarpenterNOAA", "maddenp-cu")
+
+    license("GPL-2.0-or-later", checked_by="WeirAE")
+
+    version("2.8.1", sha256="ddd306a4605f0f03e3c7b4f5d6728f9f430791effe90c2428c3e50a7aea1c165")
+    version("2.7.2", sha256="56816d543664792258bfa7dfb7e4cc66f794959dc92dc3710021f40a2b8571a4")
+    version("2.6.2", sha256="d0922ddd2b3bdbeb925c2e4694f929f3e966145d2929e74ab9f9c9ecd27b674a")
+    version("2.5.1", sha256="f389f63195492196c8009d5843a3861ad350b5fd1cea1fdb8a6bfdc7cbfd660f")
+
+    depends_on("py-pip", type="build")
+    depends_on("python@3.9:3.11")
```

Comment:
Looks like `2.8.1` can support `python@3.9:3.13`.

Reply:
And for at least `@2.5.2:2.7.2`, the range per their `meta.json` files should be `python@3.9:3.12`.

Reply:
The limit for` 3.11` was [requested](https://github.com/JCSDA/spack/pull/554#discussion_r2157855969) by JCSDA for compatibility with the unified environment. I can open it here and restrict it there if it would be preferable

Reply:
That's fine. Please add a comment with an explanation of the reason for the difference.